universum/lib/gravity.py

```diff
@@ -7,28 +9,30 @@
 
 
 class ModuleSettings:
-    def __init__(self, cls, main_settings):
+    def __init__(self, cls: Type['Module'], main_settings: 'ModuleSettings') -> None:
         object.__setattr__(self, "cls", cls)
         object.__setattr__(self, "main_settings", main_settings)
+        # TODO: can't clarify that active_modules[Type[X]] returns X (https://github.com/python/mypy/issues/4928)
+        self.active_modules: Optional[Dict[Type[Module], Module]]
 
-    def __getattribute__(self, item):
-        cls = object.__getattribute__(self, "cls")
+    def __getattribute__(self, item: Any) -> Any:  # TODO: narrow down type annotations
+        cls: Type['Module'] = object.__getattribute__(self, "cls")
         main_settings = object.__getattribute__(self, "main_settings")
         for entry in cls.__mro__:
             try:
-                settings = getattr(main_settings, entry.__name__)
+                settings: Any = getattr(main_settings, entry.__name__)  # TODO: narrow down type annotations
                 return getattr(settings, item)
             except AttributeError:
                 continue
 
         raise AttributeError("'" + cls.__name__ + "' object has no setting '" + item + "'")
 
-    def __setattr__(self, key, value):
-        cls = object.__getattribute__(self, "cls")
-        main_settings = object.__getattribute__(self, "main_settings")
+    def __setattr__(self, key: Any, value: Any) -> None:  # TODO: narrow down type annotations
```

Comment:
I couldn't easily discern whether there is any type constraint to 'key' and 'value' arguments. Please let me know if there is

Reply:
I think, as these are the settings, `key` is always `str`, and `value` may be `str` or `List[str]`. But this might require further investigation.

Reply:
thanks, I'll try these